fuels-rs/src/contract.rs

```diff
@@ -87,16 +88,37 @@ impl Contract {
         assert!(script.len() == script_len, "Script length *must* be 16");
 
         // `script_data` consists of:
-        // 1. The contract ID
-        // 2. The function selector
-        // 3. The encoded arguments, in order
+        // Primitive types as input:
+        // ┌─────────────┬─────────────┬──────────────┐
+        // │ contract_id │ fn selector │ encoded args │
+        // └─────────────┴─────────────┴──────────────┘
+        // Structs as input:
+        // ┌─────────────┬─────────────┬─────────────────┬──────────────┐
+        // │ contract_id │ fn selector │call_data_offset │ encoded args │
+        // └─────────────┴─────────────┴─────────────────┴──────────────┘
```

Comment:
Can you do something like (only did first line here because GitHub editor sucks)
```suggestion
        // ┌─────────────┬─────────────┬─────────────────┬──────────────┐
        // │ contract_id │ fn selector │call_data_offset │ encoded args │
        // └─────────────┴─────────────┴─────────────────┴──────────────┘
        //  ^^^^^^^^^^^^^ ContractId::LEN
```^

Reply:
Not sure what you mean by you doing only the first line, did you mean adding the `ContractId::LEN` to both diagrams? Like 

<img width="598" alt="image" src="https://user-images.githubusercontent.com/3484025/149384101-6e1a44c4-2f37-4f57-99d6-7f05b9c4c725.png">

Reply:
No, I meant do it for all the fields, not just contract_id. And ideally on both boxes. Updated the comment to show 2 lines.

Reply:
Uh... it was getting really unreadable like that, and I agree that we should mention the sizes. So I've changed to a simple list. Let me know what you think.